src/config/passport.ts

```diff
@@ -2,22 +2,45 @@ import GoogleTokenStategy from "passport-google-id-token";
 import JwtStrategy from "passport-jwt";
 
 import userService from "../service/user.service";
+import { UserType } from "../model/user.model";
+
+const JWT_SECRET = process.env['JWT_SECRET'] as string
 
 export const googleStrategy = new GoogleTokenStategy(
   {
     clientId: process.env.GOOGLE_CLIENT_ID,
   },
   async (parsedToken: any, googleId: any, done: any) => {
-    const foundUser = await userService.findOrCreateUserByEmail(
-      parsedToken.email
-    );
+    const randomString = (length: number) => {
+      let result = "";
+      let characters =
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+      let charactersLength = characters.length;
+      for (let i = 0; i < length; i++) {
+        result += characters.charAt(
+          Math.floor(Math.random() * charactersLength)
+        );
+      }
+      return result;
+    };
+
+    const userPassword = randomString(8);
+
+    const user: UserType = {
+      firstName: parsedToken.payload.given_name as string,
+      lastName: parsedToken.payload.family_name as string,
+      email: parsedToken.payload.email as string,
+      password: userPassword,
+      register: parsedToken.payload.iss.split('.')[1], //Anyone has any better ideas how to retrieve this field?
+    };
+    const foundUser = await userService.findOrCreateUser(user);
```

Comment:
step nay ong manually tao 1 cai password cho bon user nay kha sang tao, nhung toi van nghi minh nen de NULL cho bon user authenticate bang 3th parties. toi co list 1 vai link recommend dieu nay ong thao khao thu
- https://stackoverflow.com/questions/44524521/what-do-i-store-in-the-password-field-in-my-db-when-a-user-logs-in-with-google-o
- https://laracasts.com/discuss/channels/laravel/social-login-record-in-database-in-the-users-table-in-password-field
- https://stackoverflow.com/questions/47859468/how-to-store-user-data-once-user-login-with-facebook-google-account-on-my-websit